cg/cli/workflow/balsamic/base.py

```diff
@@ -222,85 +222,68 @@ def config_case(
             raise BalsamicStartError("No target bed specified!")
 
     # Call Balsamic
-    command_str = (
-        f" config case"
-        f" --reference-config {reference_config}"
-        f" --singularity {singularity}"
-        f" --tumor {tumor_path}"
-        f" --case-id {case_id}"
-        f" --output-config {case_id}.json"
-        f" --analysis-dir {root_dir}"
-        f" --umi-trim-length {umi_trim_length}"
-    )
-
-    if target_bed:
-        command_str += f" -p {target_bed}"
-    if normal_path:
-        command_str += f" --normal {normal_path}"
-    if umi:
-        command_str += f" --umi"
-    if quality_trim:
-        command_str += f" --quality-trim"
-    if adapter_trim:
-        command_str += f" --adapter-trim"
-    command = [f"bash -c 'source activate {conda_env}; balsamic"]
-    command_str += "'"  # add ending quote from above line
-    command.extend(command_str.split(" "))
-
-    if dry:
-        LOG.info(" ".join(command))
-        return SUCCESS
-
-    process = subprocess.run(" ".join(command), shell=True)
+    arguments = {
+        "tumor": tumor_path,
+        "normal": normal_path,
+        "case_id": case_id,
+        "output_config": f"{case_id}.json",
+        "quality_trim": quality_trim,
+        "adapter_trim": adapter_trim,
+        "umi": umi,
+        "umi_trim_length": umi_trim_length,
+        "panel_bed": target_bed,
+    }
+
+    process = context.obj["balsamic_api"].config_case(arguments=arguments, dry=dry)
     return process
 
 
 @balsamic.command()
-@click.option("-d", "--dry-run", "dry", is_flag=True, help="print command to console")
+@click.option("--dry-run", "dry", is_flag=True, help="print command to console")
 @click.option(
-    "-r", "--run-analysis", "run_analysis", is_flag=True, default=False, help="start analysis",
+    "-r", "--run-analysis", "run_analysis", is_flag=True, default=False, help="start analysis"
 )
 @click.option("--config", "config_path", required=False, help="Optional")
-@PRIORITY_OPTION
+@ANALYSIS_TYPE_OPTION
 @EMAIL_OPTION
+@PRIORITY_OPTION
 @click.argument("case_id")
 @click.pass_context
-def run(context, dry, run_analysis, config_path, priority, email, case_id):
-    """Generate a config for the case_id."""
+def run(context, dry, run_analysis, case_id, analysis_type, config_path, email, priority):
+    """Run BALSAMIC analysis for case_id."""
 
-    conda_env = context.obj["balsamic"]["conda_env"]
-    slurm_account = context.obj["balsamic"]["slurm"]["account"]
-    priority = priority if priority else context.obj["balsamic"]["slurm"]["qos"]
-    root_dir = Path(context.obj["balsamic"]["root"])
     if not config_path:
-        config_path = Path.joinpath(root_dir, case_id, case_id + ".json")
+        config_path = f'{context.obj["balsamic"]["root"]}/{case_id}/{case_id}.json'
```

Comment:
```suggestion
        config_path = Path(context.obj["balsamic"]["root"]) / case_id / (case_id + ".json")
```
I don't know about you but I suggest to always work with path objects as long as possible